ank-core/src/main/kotlin/io/kategory/ank/interpreter.kt

```diff
@@ -58,40 +62,64 @@ fun readFileImpl(source: File): String =
 fun parseMarkDownImpl(markdown: String): ASTNode =
         MarkdownParser(GFMFlavourDescriptor()).buildMarkdownTreeFromString(markdown)
 
-fun extractCodeImpl(source: String, tree: ASTNode): String {
-    val sb = StringBuilder()
+
+data class CompiledMarkdown(val origin: File, val snippets : ListKW<Snippet>)
+data class Snippet(val silent: Boolean, val startOffset: Int, val endOffset: Int, val code: String, val result: Option<String> = Option.None)
+
+fun extractCodeImpl(source: String, tree: ASTNode): ListKW<Snippet> {
+    val sb = mutableListOf<Snippet>()
     tree.accept(object : RecursiveVisitor() {
         override fun visitNode(node: ASTNode) {
             if (node.type == CODE_FENCE) {
-                sb.append("\n")
                 val fence = node.getTextInNode(source)
                 val code = fence.split("\n").drop(1).dropLast(1).joinToString("\n")
-                sb.append(code)
+                sb.add(Snippet(fence.startsWith("```$AnkSilentBlock"), node.startOffset, node.endOffset, code))
             }
             super.visitNode(node)
         }
     })
-    return sb.toString()
+    return sb.k()
 }
 
-fun compileCodeImpl(origin: File, source: String, compilerArgs: ListKW<String>): Unit {
-    val tmpFile = createTempFile(prefix = origin.name + "__", suffix = ".kt")
-    tmpFile.printWriter().use { it.println(source) }
-    K2JVMCompiler.main((
-            compilerArgs +
-                    tmpFile.absolutePath
-            ).toTypedArray())
-    tmpFile.delete()
+fun compileCodeImpl(origin: File, snippets : ListKW<Snippet>, compilerArgs: ListKW<String>): CompiledMarkdown {
+    val classLoader = URLClassLoader(compilerArgs.map {  URL(it) }.ev().list.toTypedArray())
+    val seManager = ScriptEngineManager(classLoader)
+    val engine = seManager.getEngineByExtension("kts")!!
+    val evaledSnippets = snippets.list.map { snippet ->
+        val result = engine.eval(snippet.code)
+        val resultString = Option.fromNullable(result).fold({ "// Unit" }, { "// $it" })
+        if (snippet.silent) snippet
+        else snippet.copy(result = "\n```kotlin\n$resultString\n```".some())
+    }.k()
+    return CompiledMarkdown(origin, evaledSnippets)
 }
 
-fun replaceAnkToKotlinImpl(markDown: String): String =
-        markDown.replace(AnkBlock, "kotlin")
+fun replaceAnkToKotlinImpl(compiledMarkdown: CompiledMarkdown): String {
+    val contents = mutableListOf<Char>()
+    var offset = 0
+    val filesContents = compiledMarkdown.origin.readText()
+    filesContents.toCharArray().map {
```

Comment:
should this be forEach instead? The return value isn't used and the block is side-effecty